subprojects/functional/src/test/groovy/org/gradle/internal/io/IoFunctionTest.groovy

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.io
+
+import spock.lang.Specification
+
+import static org.gradle.internal.io.IoFunction.wrap
+
+class IoFunctionTest extends Specification {
+
+    def "executed when it doesn't throw"() {
+        expect:
+        wrap({ String t -> "wrapped:$t" }).apply("lajos") == "wrapped:lajos"
+    }
+
+    def "can throw RuntimeException directly"() {
+        when:
+        wrap({ String t -> throw new RuntimeException(t) }).apply("lajos")
```

Comment:
⭕ Consider checking the identity of the thrown exception, not only the type and message.

Reply:
Done.